docs/reference-manual/native-image/DebugInfo.md

```diff
@@ -623,98 +518,64 @@ objdump --dwarf=frames hello > frames
 
 ### Currently Supported Targets
 
 The prototype is currently implemented only for the GNU Debugger on Linux:
 
-  - Linux/x86_64 support has been tested and should work
-    correctly
+  - Linux/x86_64 support has been tested and should work correctly
 
-  - Linux/AArch64 support is present but has not yet been fully
-    verified (break points should work ok but stack backtraces
-    may be incorrect)
+  - Linux/AArch64 support is present but has not yet been fully verified (break points should work ok but stack backtraces may be incorrect)
 
 Windows support is still under development.
 
 ## Debugging with Isolates
 
-Note that it is currently recommended to disable use of Isolates by
-passing flag `-H:-SpawnIsolates` on the command line when debug info
-generation is enabled. Enabling of Isolates affects the way ordinary object pointers (oops) are encoded. In turn, that means the debug info
-generator has to provide gdb with information about how to translate
-an encoded oop to the address in memory, where the object data is
-stored. This sometimes requires care when asking gdb to process
-encoded oops vs decoded raw addresses.
-
-When isolates are disabled, oops are essentially raw addresses pointing
-directly at the object contents. This is generally the same whether
-the oop is embedded in a static/instance field or is referenced from a
-local or parameter variable located in a register or saved to the stack.
-It is not quite that simple because the bottom 3 bits of some oops may
-be used to hold "tags" that record certain transient properties of
-an object. However, the debuginfo provided to gdb means that it will
-remove these tag bits before dereferencing the oop as an address.
-
-By contrast, when isolates are enabled, oops references stored in static
-or instance fields are actually relative addresses, offsets from a
-dedicated heap base register (r14 on x86_64, r29 on AArch64), rather
-than direct addresses (in a few special cases the offset may also have
-some low tag bits set). When an 'indirect' oop of this kind gets loaded
-during execution, it is almost always immediately converted to a 'raw'
-address by adding the offset to the heap base register value. So, oops
-which occur as the value of local or parameter vars are actually raw
-addresses.
-
-The DWARF info encoded into the image, when isolates are enabled, tells
-gdb to rebase indirect oops whenever it tries to dereference them to
-access underlying object data. This is normally automatic and
-transparent, but it is visible in the underlying type model that gdb
-displays when you ask for the type of objects.
-
-For example, consider the static field we encountered above. Printing
-its type in an image that uses Isolates shows that this (static) field
-has a different type to the expected one:
+Note that it is currently recommended to disable use of Isolates by passing flag `-H:-SpawnIsolates` on the command line when debug info
```

Comment:
It might be possible now to rescind this advice or, at least, to qualify it. The advice was offered because it was not possible to debug with isolates on some gdb 10 releases and all earlier releases. This has been fixed in gdb 11. So, the opening line could be modified to say this:

    Note that on some operating systems enabling the use of Isolates, by passing command line option `-H:-SpawnIsolates` to the native image generator, causes problems with printing of objects when using a `gdb` release version 10 or earlier.
    It is currently recommended to disable use of Isolates, by passing command line option `-H:-SpawnIsolates, when generating debug info if your operating system includes one of these earlier releases.
    Alternatively you may be able to upgrade your debugger to a later version.

It may also be better to move this warning from the start of this section to some later point.